src/bit_str.jl

```diff
@@ -276,210 +221,79 @@ function Base.iterate(bit::BitStr, state::Integer=1)
-swapbits(b::BitStr{T}, mask::Integer) where T = bit(swapbits(b.val, T(mask)); len=length(b))
+bfloat_r(b::BitStr{N}) where N = bfloat_r(buf(b); nbits=N)
 
 """
-    setbit(b::BitStr, mask::Integer) -> Integer
-
-set the bit at masked position to 1.
-
-# Example
-
-```jldoctest
-julia> setbit(bit"1011", 0b1100)
-1111 (15)
-
-julia> setbit(bit"1011", 0b0100)
-1111 (15)
-
-julia> setbit(bit"1011", 0b0000)
-1011 (11)
-```
+    bint_r(b::BitStr) -> Integer
 """
-setbit(b::BitStr{T}, mask::Integer) where T = setbit(b.val, T(mask)) |> bit(len=length(b))
-
+bint_r(b::BitStr{N}) where N = buf(breflect(b))
 
 """
-    anyone(b::BitStr, mask::Integer) -> Bool
-
-Return `true` if any masked position of index is 1.
-
-# Example
-
-`true` if any masked positions is 1.
-
-```jldoctest
-julia> anyone(bit"1011", 0b1001)
-true
-
-julia> anyone(bit"1011", 0b1100)
-true
-
-julia> anyone(bit"1011", 0b0100)
-false
-```
+    bint(b::BitStr) -> Integer
 """
-anyone(b::BitStr{T}, mask::Integer) where T = anyone(b.val, mask) |> bit(len=length(b))
+bint(b::BitStr) = buf(b)
 
 """
-    allone(b::BitStr, mask::Integer) -> Bool
+    bit_literal(xs...)
 
-Return `true` if all masked position of index is 1.
+Create a [`BitStr`](@ref) by input bits `xs`.
 
 # Example
 
-`true` if all masked positions are 1.
-
 ```jldoctest
-julia> allone(bit"1011", 0b1011)
-true
-
-julia> allone(bit"1011", 0b1001)
-true
-
-julia> allone(bit"1011", 0b0100)
-false
-```
-"""
-allone(b::BitStr{T}, mask::Integer) where T = allone(b.val, T(mask))
-
-"""
-    ismatch(index::Integer, mask::Integer, target::Integer) -> Bool
-
-Return `true` if bits at positions masked by `mask` equal to `1` are equal to `target`.
-
-## Example
-
-```julia
-julia> n = 0b11001; mask = 0b10100; target = 0b10000;
-
-julia> ismatch(n, mask, target)
-true
+julia> bit_literal(1, 0, 1, 0, 1, 1)
+110101 ₍₂₎
 ```
 """
-ismatch(b::BitStr{T}, mask::Integer, target::Integer) where T = ismatch(b.val, T(mask), T(target))
-
-# NOTE: we have to return a vector here since this is calculated in runtime
-
-"""
-    baddrs(b::Integer) -> Vector
-
-get the locations of nonzeros bits, i.e. the inverse operation of bmask.
-"""
-baddrs(b::BitStr) = baddrs(b.val) |> bit(len=length(b))
-
-
-"""
-    bfloat(b::BitStr) -> Float64
-
-float view, with MSB 0 bit numbering.
-See also [wiki: bit numbering](https://en.wikipedia.org/wiki/Bit_numbering)
-"""
-bfloat(b::BitStr) = bfloat(b.val; nbits=length(b))
-
-"""
-    bfloat_r(b::Integer; nbits::Int) -> Float64
-
-float view, with bits read in inverse order.
-"""
-bfloat_r(b::BitStr) = bfloat_r(b.val; nbits=length(b))
-
-"""
-    bint(b; nbits=nothing) -> Int
-
-integer view, with LSB 0 bit numbering.
-See also [wiki: bit numbering](https://en.wikipedia.org/wiki/Bit_numbering)
-"""
-bint(b::BitStr) = b.val
-
-"""
-    bint_r(b; nbits::Int) -> Integer
+bit_literal(xs...) = bit_literal(xs)
+function bit_literal(xs::NTuple{N, T}) where {N, T<:Integer}
```

Comment:
handle `BigInt` here, this can be wrong if N > 64 since we do not have assertion in the constructor.

Reply:
I don't like automatic conversion between types. I'd prefer creating another function `lbit_literal` when we really need it. Do you agree?

Reply:
yeah, then you should error here if `N > 64`